command/shell.go

```diff
@@ -154,26 +160,9 @@ func (c *shellExec) Execute(ctx context.Context,
 			c.Shell, c.Script)
 	}
 
-	if err = localCmd.Start(ctx); err != nil {
-		logger.System().Debugf("error spawning shell process: %v", err)
-		return err
-	}
-
-	pid := localCmd.GetPid()
-
-	logger.System().Debugf("spawned shell process with pid %d", pid)
-
-	// Call the platform's process-tracking function. On some OSes this will be a noop,
-	// on others this may need to do some additional work to track the process so that
-	// it can be cleaned up later.
-	subprocess.TrackProcess(conf.Task.Id, pid, logger.System())
```

Comment:
Where do these procs get cleaned up?

Reply:
We're still calling the legacy TrackProcess code (it's just called in the process constructor.

The cleanup, as usual, is in the agent/inter (and pre) task running.